src/extract.c

```diff
@@ -5,48 +5,124 @@
 #include "backports.h"
 #include <string.h>
 
-int find_offset(SEXP x, SEXP index, int i) {
+int find_offset(SEXP x, SEXP index, int i, int strict) {
   if (Rf_length(index) > 1) {
-    Rf_errorcall(R_NilValue, "Index %i must have length 1", i + 1);
+    Rf_errorcall(R_NilValue,
+      "Index %i must have length 1, not %i.",
```

Comment:
I think these messages are a bit easier to write/read if they all start in the same way, either with "Invalid object:" or "Invalid index (%i):".